src/api/app/models/repository_architecture.rb

```diff
@@ -10,6 +10,8 @@ class RepositoryArchitecture < ApplicationRecord
   validates :repository, uniqueness: { scope: :architecture }
 
   def build_id
+    return if required_checks.empty?
```

Comment:
I find this problematic. There still is a build_id, so I would avoid calling the function, but not have the function fake there is no build_id

Reply:
What I'd suggest is (untested):

```
index 4c43029e7..04f8efe80 100644
--- a/src/api/app/models/project.rb
+++ b/src/api/app/models/project.rb
@@ -1675,6 +1675,7 @@ class Project < ApplicationRecord
   end
 
   def status_reports(checkables)
+    checkables = checkables.reject { |checkable| checkable.required_checks.empty? }
     status_reports = Status::Report.where(checkable: checkables)
     result = {}
     status_reports.where(uuid: checkables.map(&:build_id)).find_each do |report|

Reply:
Makes sense.